addons/block_code/ui/blocks/statement_block/statement_block.gd

```diff
@@ -14,9 +14,6 @@ var param_input_strings: Dictionary  # Only loaded from serialized
 
 func _ready():
 	super()
-
-	if block_type != Types.BlockType.EXECUTE:
-		_background.show_top = false
 	_background.color = color
```

Comment:
This goes back to the thing about statement blocks able to be used for multiple purposes. Maybe the real rename we should do is changing `StatementBlock` to `BlockWithParametersThatCanFormatStringToCode` or something lol. I think we should keep these lines in because they allow a statement block to not just be an execute block, meaning they have no snapping notch at the top.

Reply:
In my view "StatementBlock" means presicely that, a block that translate to a line of GDScript.

Reply:
Yes, I agree, the general purpose of `StatementBlock` is to generate a line of GDScript, but I had recently used it to implement these entry blocks with parameters. Seemed silly to make an entire new block class to do something I already could do with the `StatementBlock`, that's why I am thinking it could be renamed to something like `ParameterInputsBlock` or `ParameterFormatBlock` to cover it's general functionality.

Reply:
Well, "On process" will need to expose the delta parameter at some point too. In my mind, it should be an Entry block, not a StatementBlock/ParameterInputsBlock considering Entry blocks as (from my spec):

- Entry block:
	- Translates to a Godot method, which may be virtual a method like `_ready()`.
		- Thus, may have parameters, each one with its own Variant type.
	- Can't be placed inside other blocks slots (it's root)
	- Has one slot.
		- If nothing is inside, the method body translates to `pass`.
	- Only one possible. Ie. if there are 2 "On process" blocks in the canvas, one is greyed out.
	- Block parameters can be used only in their body blocks:
		- Can drag "body" from "On body [body] entered" into an "if node [] is in group ["foo"]"
		- Can't drag "body" outside this set of blocks.

Reply:
That makes sense. You can do all of the functionality you describe in the spec with a `StatementBlock` but if you want to create another `EntryBlock` class that shares much of its functionality with `StatementBlock` you could (look at `ParameterBlock` for example).

However, it does seem like the easiest option is to just rename `StatementBlock` to `ParameterInputsBlock` and then when actually constructing the entry blocks, set their `block_type` to `ENTRY`. This is what is currently happening.